evap/staff/views.py

```diff
@@ -1033,21 +1084,39 @@ def user_create(request):
 
 @staff_required
 def user_import(request):
-    form = UserImportForm(request.POST or None, request.FILES or None)
-    operation = request.POST.get('operation')
+    excel_form = UserImportForm(request.POST or None, request.FILES or None)
 
-    if form.is_valid():
+    if request.method == "POST":
+        operation = request.POST.get('operation')
         if operation not in ('test', 'import'):
             raise SuspiciousOperation("Invalid POST operation")
 
         test_run = operation == 'test'
-        excel_file = form.cleaned_data['excel_file']
-        UserImporter.process(request, excel_file, test_run)
-        if test_run:
-            return render(request, "staff_user_import.html", dict(form=form))
-        return redirect('staff:user_index')
+        import_run = operation == 'import'
+
+        if test_run and excel_form.is_valid():
+            excel_file = excel_form.cleaned_data['excel_file']
+            _, warnings, errors = UserImporter.process(request, excel_file, test_run)
```

Comment:
`__` please, `_` is an import, i don't want to know in how many ways this can break things :)